x-pack/plugins/fleet/public/components/agent_enrollment_flyout/hooks.tsx

```diff
@@ -74,19 +78,43 @@ export function useIsK8sPolicy(agentPolicy?: AgentPolicy) {
 }
 
 export function useCloudSecurityIntegration(agentPolicy?: AgentPolicy) {
+  const integrationVersion = useMemo(() => {
+    return getCloudSecurityIntegrationVersionFromAgentPolicy(agentPolicy);
+  }, [agentPolicy]);
+
+  const { data: packageInfoData, isLoading } = useGetPackageInfoByKeyQuery(
```

Comment:
Is the package now being queried every time we open the Add Agent flyout? I am wondering if we could optimize here to only query if needed.

Reply:
yes, great catch actually, I'll optimize it to query only when it's detected a Cloud Security integration, thanks

Reply:
hi @juliaElastic, I added a logic that updates the hook to only fetch data when it detects a Cloud Security Integration, and also `react-query` has its own cache, so it won't query every time the Agent Flyout anymore. It uses the `enabled` option of the `useQuery` hook to achieve that.